src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -903,14 +903,27 @@ public virtual void Update(IndexDefinitionBase definition, IndexingConfiguration
 
                 _indexStorage.WriteDefinition(definition);
 
+                if (definition.LastStateChangeRaftIndex != Definition.LastStateChangeRaftIndex)
+                {
+                    if (definition.State == IndexState.Normal)
+                    {
+                        SetState(IndexState.Normal);
+                        stateChanged = true;
+                    }
+                    else
+                    {
+                        SetState(definition.State);
+                    }
+                }
+
                 Definition = definition;
                 Configuration = configuration;
 
                 OnInitialization();
 
                 _priorityChanged.Raise();
 
-                if (status == IndexRunningStatus.Running)
+                if ((status == IndexRunningStatus.Running) || ((stateChanged) && (definition.State == IndexState.Normal)))
```

Comment:
why `((status == IndexRunningStatus.Running) || stateChanged)` isn't enough here?

Reply:
the state can be changed to disabled

Reply:
but we set it to true only if `definition.State == IndexState.Normal`, or do I miss something?

Reply:
yes, you are right